indico/modules/events/layout/controllers/layout.py

```diff
@@ -65,12 +69,50 @@ def _process_lecture_meeting(self):
             return redirect(url_for('.index', self.event_new))
         return WPLayoutEdit.render_template('layout_meeting_lecture.html', self._conf, form=form, event=self.event_new)
 
+    def _make_weeks_view_form(self):
+        form_class = type(b'WeeksViewForm', (IndicoForm,), {})
+        fields = theme_settings.themes['indico-weeks-view']['arguments']
+        defaults = {}
+        for field_name in fields:
+            field_type = fields[field_name]['type']
+            field_class = getattr(indico_fields, field_type, None) or getattr(wtforms_fields, field_type, None)
+            if not field_class:
+                continue
+            field_object = type(str(field_name), (field_class,), {})
```

Comment:
You don't need to subclass the field class. Just create an instance of `field_class` below...